repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -118,6 +120,24 @@ private void processUpdateProduct(AtlasEntity entity, AtlasVertex vertex) throws
             throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "DataProduct can only be moved to another Domain.");
         }
 
+        if(entity.getAttribute(DAAP_LINEAGE_STATUS_ATTR) != null && entity.getAttribute(DAAP_LINEAGE_STATUS_ATTR).equals(DAAP_LINEAGE_STATUS_COMPLETED)){
+            if (!ARGO_SERVICE_USER_NAME.equals(RequestContext.getCurrentUser())) {
+                throw new AtlasBaseException(AtlasErrorCode.UNAUTHORIZED_ACCESS, RequestContext.getCurrentUser(), "Lineage Status Update");
+            }
+
+            if (vertex.getProperty(DAAP_LINEAGE_STATUS_ATTR, String.class).equals(DAAP_LINEAGE_STATUS_IN_PROGRESS)){
+                entity.setAttribute(DAAP_LINEAGE_STATUS_ATTR, DAAP_LINEAGE_STATUS_COMPLETED);
+            }
+        }
+
+        if(entity.hasAttribute(DAAP_ASSET_DSL_ATTR)) {
+            entity.setAttribute(DAAP_LINEAGE_STATUS_ATTR, DAAP_LINEAGE_STATUS_PENDING);
+        }
+
+        if(entity.hasAttribute(STATE_PROPERTY_KEY)){
```

Comment:
Same here, If entity.STATE_PROPERTY_KEY value is same what asset currently has, it will create unnecessary update event for Lineage caluculations